planner/core/logical_plan_builder.go

```diff
@@ -197,6 +198,108 @@ func (a *aggOrderByResolver) Leave(inNode ast.Node) (ast.Node, bool) {
 	return inNode, true
 }
 
+func (b *PlanBuilder) buildExpand(p LogicalPlan, gbyItems []expression.Expression) (LogicalPlan, []expression.Expression, error) {
+	b.optFlag |= flagResolveExpand
+
+	// Rollup syntax require expand OP to do the data expansion, different data replica supply the different grouping layout.
+	distinctGbyExprs, gbyExprsRefPos := expression.DeduplicateGbyExpression(b.ctx, gbyItems)
+	// build another projection below.
+	proj := LogicalProjection{Exprs: make([]expression.Expression, 0, p.Schema().Len()+len(distinctGbyExprs))}.Init(b.ctx, b.getSelectOffset())
+	// project: child's output and distinct GbyExprs in advance. (make every group-by item to be a column)
+	projSchema := p.Schema().Clone()
+	names := p.OutputNames()
```

Comment:
Why do we need to keep the original column in the projection node here?

Reply:
just as the comment said: 
// project: child's output and distinct GbyExprs in advance. (make every group-by item to be a column)

Expand shouldn't change the child's output, it only appends some copy-necessary cols and generated columns like gid and gpos